frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form/ocs-storage-class-form.tsx

```diff
@@ -161,3 +189,264 @@ export const PoolResourceComponent: React.FC<ProvisionerProps> = ({ onParamChang
+
+  const csiCMWatchResource: WatchK8sResource = {
+    kind: ConfigMapModel.kind,
+    namespaced: true,
+    isList: false,
+    namespace: CEPH_STORAGE_NAMESPACE,
+    name: KMSConfigMapCSIName,
+  };
+
+  const [csiConfigMap, csiConfigMapLoaded, csiConfigMapLoadError] = useK8sWatchResource<
+    ConfigMapKind
+  >(csiCMWatchResource);
+
+  React.useEffect(() => {
+    if (!_.isEmpty(csiConfigMap)) {
+      const serviceNames = Object.keys(csiConfigMap?.data);
+      const kmsData = JSON.parse(csiConfigMap?.data[serviceNames[serviceNames.length - 1]]);
+      setCurrentKMS(kmsData);
+      const url = parseURL(kmsData.VAULT_ADDR);
+      const kmsObj: KMSConfig = {
+        name: {
+          value: kmsData.KMS_SERVICE_NAME,
+          valid: true,
+        },
+        address: {
+          value: `${url.protocol}//${url.hostname}`,
+          valid: true,
+        },
+        port: {
+          value: url.port,
+          valid: true,
+        },
+        backend: kmsData.VAULT_BACKEND_PATH,
+        caCert: state.kms.caCert ?? null,
+        caCertFile: state.kms.caCertFile,
+        tls: kmsData.VAULT_TLS_SERVER_NAME,
+        clientCert: state.kms.clientCert ?? null,
+        clientCertFile: state.kms.clientCertFile,
+        clientKey: state.kms.caCert ?? null,
+        clientKeyFile: state.kms.clientKeyFile,
+        providerNamespace: kmsData.VAULT_NAMESPACE,
+        hasHandled: true,
+      };
+      setEncryptionDispatch(SCActionType.SET_KMS_ENCRYPTION, '', dispatch, kmsObj);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [csiConfigMap, editKMS]);
+
+  React.useEffect(() => {
+    if (editKMS) {
+      if (state.kms.name.valid && state.kms.address.valid && state.kms.port.valid) {
+        setValidSave(true);
+      } else {
+        setValidSave(false);
+      }
+      setEncryptionDispatch(SCActionType.SET_KMS_ENCRYPTION, '', dispatch, state.kms);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [
+    state.kms.name.value,
+    state.kms.address.value,
+    state.kms.port.value,
+    state.kms.name.valid,
+    state.kms.address.valid,
+    state.kms.port.valid,
+  ]);
+
+  React.useEffect(() => {
+    onParamChange(checked.toString());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [checked]);
+
+  const StorageClassEncryptionLabel: React.FC = () => (
+    <div className="ocs-storageClass-encryption__pv-title">
+      <span className="ocs-storageClass-encryption__pv-title--padding">Enable Encryption</span>
+      <TechPreviewBadge />
+    </div>
+  );
+
+  const EncryptionCardLoading: React.FC = () => (
+    <Grid className="skeleton-box">
+      <GridItem span={12} className="skeleton-activity" />
+      <GridItem span={12} className="skeleton-activity" />
+      <GridItem span={12} className="skeleton-activity" />
+      <GridItem span={12} className="skeleton-activity" />
+    </Grid>
+  );
```

Comment:
Components dont need to be inside existing component.
These should be moved out